client/tests/test_zksnark.py

```diff
@@ -8,8 +8,321 @@
+}
+
+PROOF_BW6_761_GROTH16_PARAMETERS = [
+    # "a":
+    [
+        int("00b42fc65c4178e23c5ea46791b63f13e01057d957d097d2a7b1b99b921b3db0", 16),  # noqa
+        int("b519b21bd21f9d5209420de0d39e6ceebcf40df23e8f3dfb3544e3f221687a25", 16),  # noqa
+        int("4f935e7e4eafbded993af4464cf7ca8da374b2cbcc6003fb47bc590dd8eaadc2", 16),  # noqa
+        int("001f63f85f5e96168363e1c3733094347b9d7d0cbb2b762c65c12b52fe92e126", 16),  # noqa
+        int("b1f884d331d7b8740dccb383d7565eeb625fc43598bd371801153e0a690e1881", 16),  # noqa
+        int("f84849653fce01034cb571b78232b5e7aab22f0b3ee089c0b907de8a52628a92", 16),  # noqa
+    ],
+    # "b":
+    [
+        int("00bfb5be9eb134d7118ab1f759b5a801dda03315108848082a6815dab0c88fe2", 16),  # noqa
+        int("53429d65b7b03a7983a6ee353f0f9687de39888afe4fcb106900a10cee2c4c42", 16),  # noqa
+        int("d6efa2ee7cdc8d82b052fa8e0f79786d2a4847a25d9ca9026a106de6c73c8d18", 16),  # noqa
+        int("00b9f29ad8d2107e760fa728a897b26b673e3b099e56e7c2bdfe0194cd02f8af", 16),  # noqa
+        int("f4b799f6f8d07f6e3b7dfc000e02eda978e1993a57337b5e2f2e9e3f024ef303", 16),  # noqa
+        int("67887ed23cca57cc33d8bfafdfb4c914e085870621cf02bb380b80387162fb40", 16),  # noqa
+    ],
+    # "c":
+    [
+        int("003f75f402703fb7d597cd9beb33fb216af606a687c133ef8b73fde17a48c12b", 16),  # noqa
+        int("e3f17867679ccd5958ceb9245adac2377eb1444c6577049f04c0a18645b00a4b", 16),  # noqa
+        int("ae9c6274cd8876f52f5307dfc50935b5f515ee33c5e98031705fe4ce153da553", 16),  # noqa
+        int("00c83d865b8c18f4120fdc9f45026e252d05ceb3f0dfcd19a8e2f11d2a8cd6cd", 16),  # noqa
+        int("b7450c0fc8e0b1a284db1c21d25d9fbea91d741713f414f577ccb8455e1c55af", 16),  # noqa
+        int("07b72c4868e58c9890c0335bf13a5821391f0cc8c38ab1f168314f1cb67b10a1", 16),  # noqa
+    ]
+]
+# pylint: enable=line-too-long
+
+
 class TestZKSnark(TestCase):
 
+    def test_bls12_377_groth16_verification_key_parameters(self) -> None:
+        vk = VERIFICATION_KEY_BLS12_377_GROTH16
+        vk_parameters_expect = VERIFICATION_KEY_BLS12_377_GROTH16_PARAMETERS
+        vk_parameters = \
+            zksnark.Groth16SnarkProvider().verification_key_to_evm_parameters(vk)
+        self.assertEqual(vk_parameters_expect, vk_parameters)
+
+    def test_bls12_377_groth16_proof_parameters(self) -> None:
+        extproof = EXTPROOF_BLS12_377_GROTH16
+        proof_parameters = \
+            zksnark.Groth16SnarkProvider().proof_to_evm_parameters(extproof)
+        self.assertEqual(PROOF_BLS12_377_GROTH16_PARAMETERS, proof_parameters)
+
+    def test_bw6_761_groth16_verification_key_parameters(self) -> None:
+        vk = VERIFICATION_KEY_BW6_761_GROTH16
+        vk_parameters_expect = VERIFICATION_KEY_BW6_761_GROTH16_PARAMETERS
+        vk_parameters = \
+            zksnark.Groth16SnarkProvider().verification_key_to_evm_parameters(vk)
+        self.assertEqual(vk_parameters_expect, vk_parameters)
+
+    def test_bw6_761_groth16_proof_parameters(self) -> None:
```

Comment:
I can imagine that this test (BW6) is here for completeness as `libzeth` should be generic enough and re-used, especially in Zecale, where BW6 is relevant (no Zeth proof will be generated over BW6). As such, I'd add a test case for the other supported curve, namely `bn254`, for completeness, even if we know that the elements of the associated fields (base/scalar) fit in a single EVM word.

Reply:
Added tests for alt_bn128.

Reply:
Thanks